### PR TITLE
StoreGateway: Add a metric to track block load duration

### PR DESCRIPTION
**Pull request by harry671003**
* [ ] I added CHANGELOG entry for this change.
* [X] Change is not relevant to the end user.

## Changes
- Add a new histogram to track the store-gateway block sync duration.

### CODE REVIEW

**Review by yeya24 — DISMISSED** [review]
Generally I am ok with this, hopefully I can get another approval before I merge it

File: pkg/store/bucket.go
Comment (yeya24) [inline pkg/store/bucket.go:718]: Do we want to record the time of mainly `NewBinaryReader` function?
This will include cases where we hit the local cache and not load index header from object storage.

Do we have a better way to track the block load duration? But I am fine to leave it as it is. P99 can usually tell the case where we need to download remotely.

**Comment by yeya24** [comment]
We need to fix linter first

File: pkg/store/bucket.go
Comment (harry671003) [reply pkg/store/bucket.go:718]: I could add another metric for specifically measuring the time taken for `WriteBinary()`.

**Review by fpetkovski — APPROVED** [review]

**Pull request merged**
